scopes/workspace/workspace/bit-map.ts

```diff
@@ -192,14 +192,15 @@ export class BitMap {
       const config = componentMap.config;
       if (!config) return;
       Object.keys(config).forEach((aspectId) => {
-        if (aspectId === sourceId.toString()) {
+        const fullSourceId = this.getBitmapEntry(sourceId).id.toStringWithoutVersion();
+        if (aspectId === fullSourceId) {
           config[targetId.toString()] = config[aspectId];
           delete config[aspectId];
           this.markAsChanged();
         }
         if (aspectId === EnvsAspect.id) {
           const envConfig = config[aspectId];
-          if (envConfig !== REMOVE_EXTENSION_SPECIAL_SIGN && envConfig.env === sourceId.toString()) {
+          if (envConfig !== REMOVE_EXTENSION_SPECIAL_SIGN && envConfig.env !== sourceId.toString()) {
```

Comment:
Why this need be!== and not === 
With!== it will replace any env for any component to the new one

Reply:
because it works?   🤣 
envConfig.env is the env prop in the config for the currently iterated component in the bitmap.
the sourceId (should change it to the fullSoruceId) is the env we want to change.
So you are right and I have to fix it now 😨 (I was testing it on one component)

Fixed on next commit. Also fixed a case where the source env is specified with a version and it wouldn't detect it.